diff -Naur a/src/config.c b/src/config.c
--- a/src/config.c	2013-08-11 08:05:28.389665421 +0100
+++ b/src/config.c	2013-08-11 08:07:43.268139960 +0100
@@ -579,7 +579,7 @@
 	else if (strcmp(key, "launcher_icon_size") == 0) {
 		launcher_max_icon_size = atoi(value);
 	}
-	else if (strcmp(key, "launcher_item_app") == 0) {
+	else if (strncmp(key, "launcher_item_app", 17) == 0) {
 		char *app = strdup(value);
 		panel_config.launcher.list_apps = g_slist_append(panel_config.launcher.list_apps, app);
 	}
