diff -Naur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2013-07-28 00:38:52.030341757 +0100
+++ b/CMakeLists.txt	2013-07-28 00:38:45.670505144 +0100
@@ -61,9 +61,9 @@
              src/util/window.c )
 
 option( ENABLE_BATTERY "Enable battery status plugin" ON )
-option( ENABLE_TINT2CONF "Enable tint2conf build, a GTK+2 theme switcher for tint2" ON )
+option( ENABLE_TINT2CONF "Enable tint2conf build, a GTK+2 theme switcher for tint2" OFF )
 option( ENABLE_EXAMPLES "Install additional tin2rc examples" OFF )
-option( ENABLE_SN "Startup notification support" ON )
+option( ENABLE_SN "Startup notification support" OFF )
 if( ENABLE_SN )
   if( SN_FOUND )
     add_definitions( -DHAVE_SN -DSN_API_NOT_YET_FROZEN )
@@ -114,10 +114,12 @@
 set_target_properties(tint2 PROPERTIES LINK_FLAGS "-pthread" )
 
 install( TARGETS tint2 DESTINATION bin )
-install( FILES sample/tint2rc DESTINATION ${SYSCONFDIR}/xdg/tint2 )
-install( FILES default_icon.png DESTINATION ${DATADIR}/tint2 )
-install( FILES AUTHORS ChangeLog README DESTINATION ${DOCDIR} )
-install( FILES doc/tint2.1 DESTINATION ${MANDIR}/man1 )
+# install( FILES sample/tint2rc DESTINATION ${SYSCONFDIR}/xdg/tint2 )
+# install( FILES default_icon.png DESTINATION ${DATADIR}/tint2 )
+# install( FILES AUTHORS ChangeLog README DESTINATION ${DOCDIR} )
+# install( FILES doc/tint2.1 DESTINATION ${MANDIR}/man1 )
+install( FILES src/tint2conf/tintwizard.py DESTINATION bin )
+
 if( ENABLE_EXAMPLES )
   file( GLOB SAMPLEFILES sample/*.tint2rc )
   install( FILES ${SAMPLEFILES} DESTINATION ${DOCDIR}/examples )
