From b74c0765335d460ae84549f5977ffed588a26c23 Mon Sep 17 00:00:00 2001
From: David-John Willis <John.Willis@Distant-earth.com>
Date: Sun, 8 Feb 2009 18:58:35 +0000
Subject: [PATCH] Add missing define to EHCI-OMAP.c

---
 drivers/usb/host/ehci-omap.c |    7 +++++--
 1 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/usb/host/ehci-omap.c b/drivers/usb/host/ehci-omap.c
index 3c01a91..b3c1c35 100755
--- a/drivers/usb/host/ehci-omap.c
+++ b/drivers/usb/host/ehci-omap.c
@@ -50,9 +50,12 @@
 #define EXTERNAL_PHY_RESET
 //#define	EXT_PHY_RESET_GPIO_PORT1	(57)
 //#define	EXT_PHY_RESET_GPIO_PORT1	(147)
-#define	EXT_PHY_RESET_GPIO_PORT1	(16)
+#define	EXT_PHY_RESET_GPIO_PORT1		(16)
 //#define	EXT_PHY_RESET_GPIO_PORT2	(61)
-#define	EXT_PHY_RESET_DELAY		(10)
+#define	EXT_PHY_RESET_DELAY				(10)
+
+#define PHY_STP_PULLUP_ENABLE           (0x10)
+#define PHY_STP_PULLUP_DISABLE          (0x90)
 
 /* ISSUE2:
  * USBHOST supports External charge pump PHYs only
-- 
1.6.0.5

