--- orig/src/camera.h	2013-07-16 17:45:30.568780832 +0000
+++ pygame-1.9.1release/src/camera.h	2013-07-16 17:52:43.107115790 +0000
@@ -39,7 +39,6 @@
 
     #include <asm/types.h>          /* for videodev2.h */
 
-    #include <linux/videodev.h>
     #include <linux/videodev2.h>
 #endif
 
@@ -51,7 +50,6 @@
 #define RGB_OUT 1
 #define YUV_OUT 2
 #define HSV_OUT 4
-#define CAM_V4L 1
 #define CAM_V4L2 2
 
 struct buffer 
@@ -111,8 +109,4 @@
 int v4l2_close_device (PyCameraObject* self);
 int v4l2_open_device (PyCameraObject* self);
 
-/* internal functions specific to v4l */
-int v4l_open_device (PyCameraObject* self);
-int v4l_init_device(PyCameraObject* self);
-int v4l_start_capturing(PyCameraObject* self);
 #endif
--- orig/src/_camera.c	2013-07-16 17:45:30.565447459 +0000
+++ pygame-1.9.1release/src/_camera.c	2013-07-16 18:01:55.079807999 +0000
@@ -160,20 +160,6 @@
 {
 #if defined(__unix__)
     if (v4l2_open_device(self) == 0) {
-        if (v4l_open_device(self) == 0) {
-            v4l2_close_device(self);
-            return NULL;
-        } else {
-            self->camera_type = CAM_V4L;
-            if (v4l_init_device(self) == 0) {
-                v4l2_close_device(self);
-                return NULL;
-            }
-            if (v4l_start_capturing(self) == 0) {
-                v4l2_close_device(self);
-                return NULL;
-            }
-        }
     } else {
         self->camera_type = CAM_V4L2;
         if (v4l2_init_device(self) == 0) {
--- orig/Setup.in	2013-07-16 17:45:30.578780952 +0000
+++ pygame-1.9.1release/Setup.in	2013-07-16 18:25:30.392338720 +0000
@@ -34,7 +34,7 @@
 _numericsndarray src/_numericsndarray.c $(SDL) $(MIXER) $(DEBUG)
 movie src/movie.c $(SDL) $(SMPEG) $(DEBUG)
 scrap src/scrap.c $(SDL) $(SCRAP) $(DEBUG)
-_camera src/_camera.c src/camera_v4l2.c src/camera_v4l.c $(SDL) $(DEBUG)
+_camera src/_camera.c src/camera_v4l2.c $(SDL) $(DEBUG)
 pypm src/pypm.c $(SDL) $(PORTMIDI) $(PORTTIME) $(DEBUG)
 
 GFX = src/SDL_gfx/SDL_gfxPrimitives.c 
