diff -Naur a/menu-cache-gen/desktop-entries.c b/menu-cache-gen/desktop-entries.c
--- a/menu-cache-gen/desktop-entries.c	2013-07-31 06:26:05.871188801 +0100
+++ b/menu-cache-gen/desktop-entries.c	2013-07-31 06:25:29.764125135 +0100
@@ -311,7 +311,14 @@
     g_error_free (err);
   }
 */
-  retval->comment      = GET_LOCALE_STRING ("Comment");
+   /* Fix for Open Pandora's codec pack, which has newline char in Comment field.
+   * This pushes everything else (like icon and exec patch) down into the wrong slot.
+   * I could parse every comment for  '\n' and nuke it, but it's not like anything
+   * is making active use of .desktop file comments, so I'm just going to replace it
+   * with the name
+   */
+  /* retval->comment      = GET_LOCALE_STRING ("Comment"); */
+  retval->comment         = GET_LOCALE_STRING ("Name");
   retval->icon         = GET_LOCALE_STRING ("Icon");
   retval->flags        = get_flags_from_key_file (retval, key_file, desktop_entry_group);
   retval->categories   = get_categories_from_key_file (retval, key_file, desktop_entry_group);
