Index: git/Makefile
===================================================================
--- git.orig/Makefile	2009-12-07 10:01:27.091137735 +0000
+++ git/Makefile	2009-12-07 10:02:52.995649783 +0000
@@ -11,18 +11,18 @@
 RM = rm
 
 # environment
-LDFLAGS = -L$(PNDSDK)/usr/lib -lSDL -lSDL_image -lSDL_ttf -lfreetype -lGLES2D \
+EXTRALDFLAGS = -L$(PNDSDK)/usr/lib -lSDL -lSDL_image -lSDL_ttf -lfreetype -lGLES2D -lconfig \
 		-lGLES_CM -lIMGegl -lsrv_um -lXau -lXdmcp -lXi -lXext -lX11 \
 		-lpnd -lstdc++
 
-CFLAGS = -Wall -Werror -g -Ilibconfig -Ilibpnd -I$(PNDSDK)/usr/include
+CFLAGS += -Wall -Werror -g -I$(PNDSDK)/usr/include
 CFLAGS += -O3 -march=armv7-a -mcpu=cortex-a8 -mtune=cortex-a8 -mfloat-abi=softfp \
 		-mfpu=neon -ftree-vectorize -ffast-math -fomit-frame-pointer \
 		-fno-strict-aliasing -fsingle-precision-constant
 
 # code
 TARGET = pmenu
-LIBCONFIG = libconfig/.libs/libconfig.a
+
 ALLOBJ = category_media.o category_settings.o config_favourite.o \
 	config_pmenu.o config_skin.o get_apps.o main.o utils.o \
 	utils_cpu.o utils_mplayer.o
@@ -38,21 +38,13 @@
 	cd libconfig; make distclean; cd ..;
 	find . -name "*~*" -exec rm {} \; -print
 
-
 # component targets
 #
 
-pmenu:	${LIBCONFIG} ${ALLOBJ}
-	${CC} ${ALLOBJ} ${LIBCONFIG} ${LDFLAGS} -o ${TARGET}
+pmenu:	${ALLOBJ}
+	${CC} ${ALLOBJ} ${EXTRALDFLAGS} -o ${TARGET}
 	${STRIP} ${TARGET}
 
-libconfig/.libs/libconfig.a:
-	cd libconfig; make distclean; \
-	LDFLAGS="-L$PNDSDK/usr/lib" CFLAGS="-I$PNDSDK/usr/include" \
-		./configure --host=arm-none-linux-gnueabi --target=arm-none-linux-gnueabi; \
-	make libconfig.la; \
-	cd ..;
-
 # deployment and assembly components
 #
 
